File: dotnet/System/Workspace/Allors.Workspace.Client/Request/Contains.cs
// <copyright file="Contains.cs" company="Allors bvba">
// Copyright (c) Allors bvba. All rights reserved.
// Licensed under the LGPL license. See LICENSE file in the project root for full license information.
// </copyright>

namespace Allors.Workspace.Request
{
    using Allors.Workspace.Meta;
    using Allors.Workspace.Request.Visitor;
    using Allors.Workspace.Response;

    public class Contains : IPropertyPredicate
    {
        public Contains(IRelationEndType relationEndType = null) => this.RelationEndType = relationEndType;

        public IObject Object { get; set; }

        public string Parameter { get; set; }

        public IRelationEndType RelationEndType { get; set; }

        public void Accept(IVisitor visitor) => visitor.VisitContains(this);
    }
}

File: typescript/modules/libs/system/workspace/domain/src/lib/data/contains.ts
import { RelationEndType } from '@allors/system/workspace/meta';
import { IObject } from '../iobject';
import { ParameterizablePredicateBase } from './parameterizable-predicate';

export interface Contains extends ParameterizablePredicateBase {
  kind: 'Contains';
  relationEndType: RelationEndType;
  object?: IObject;
  objectId?: number;
}
